lib/pry/core_extensions.rb

```diff
@@ -103,9 +105,9 @@ def __binding__
         # class_eval sets the default definee to self.class
         self.class.class_eval(*Pry::BINDING_METHOD_IMPL)
       end
+      retry
```

Comment:
Could this result in infinite loop?

Reply:
A `__pry__` method that raises isn't our `__pry__` method. So, we define our `__pry__` and retry. This time it's our `__pry__` so it will not raise.